app/controllers/webhooks/mailchimp_unsubscribes_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Webhooks::MailchimpUnsubscribesController < ApplicationController
+  class InvalidListID < StandardError; end
+
+  LIST_MAPPINGS = {
+    mailchimp_newsletter_id: :email_newsletter,
+    mailchimp_sustaining_members_id: :email_membership_newsletter,
+    mailchimp_tag_moderators_id: :email_tag_mod_newsletter,
+    mailchimp_community_moderators_id: :email_community_mod_newsletter
+  }.freeze
+
+  def create
+    not_authorized unless valid_secret?
+    user = User.find_by!(email: params.dig(:data, :email))
+    user.update(email_type => false)
+  end
+
+  private
+
+  def valid_secret?
+    params[:secret] == SiteConfig.mailchimp_webhook_secret
+  end
+
+  def email_type
+    list_id = params.dig(:data, :list_id)
+    config = LIST_MAPPINGS.keys.detect { |k| SiteConfig.public_send(k) == list_id }
```

Comment:
This is slightly awkward, but `SiteConfig` stores the values as YAML and looking them up with `SiteConfig.where(value: params.dig(:data, :list).to_yaml.chomp` isn't exactly great either.

Reply:
This is great. I can't think of a better way to do this either.